client/src/Components/Layout/Login/Login.jsx

```diff
@@ -0,0 +1,331 @@
+  Visibility,
+  VisibilityOff,
+} from '@material-ui/icons';
+import Cookies from 'js-cookie';
+import { withStyles } from '@material-ui/core/styles';
+
+import LoaderProgress from '../../Common/LoaderProgress';
+
+import axios from 'axios';
+import LoginStyles from './LoginStyle';
+
+class Login extends Component {
+  state = {
+    logindata: { email: null, password: null },
+    isLoading: false,
+    displayBlock: true,
+    showPassword: false,
+    email: {
+      Message: '',
+      isValid: true,
+    },
+    password: {
+      Message: '',
+      isValid: true,
+      minLength: 3,
+    },
+    redirect: false,
+    ReturnUrlText: '/',
+    SignupLink: '/user/Signup/',
+  };
+
+  componentDidMount() {
+    const AuthToken = Cookies.get('AuthToken');
+
+    console.log(AuthToken);
+    console.log(!AuthToken);
+    const qstring = queryString.parse(this.props.history.location.search);
+    if (qstring.ReturnUrl) {
+      this.setState({
+        ReturnUrlText: qstring.ReturnUrl,
+        SignupLink: `/user/Signup/?ReturnUrl=${qstring.ReturnUrl}`,
+      });
+    } else {
+      this.setState({
+        // if the user log in we will redirect to Home
+        redirect: !AuthToken ? false : true,
+      });
+    }
+  }
+
+  renderAction = () => {
+    if (this.state.redirect) {
+      return <Redirect to={this.state.ReturnUrlText} />;
+    }
+  };
+
+  texthandler = (e) => {
+    const form = this.state.logindata;
+    form[e.target.name] = e.target.value;
+    this.setState({ logindata: form });
+  };
+
+  handleClickShowPassword = () => {
+    const showPassword = this.state.showPassword;
+
+    this.setState({
+      showPassword: !showPassword,
+    });
+  };
+
+  handleMouseDownPassword = (event) => {
+    event.preventDefault();
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const { email, password, logindata } = this.state;
+
+    if (!logindata.email) {
+      email.Message = 'The Email is Required';
+      email.isValid = false;
+    } else if (!validateEmail(logindata.email)) {
+      // the Email is not Valid
+      email.Message = 'Please Enter Valid Email';
+      email.isValid = false;
+    } else {
+      email.Message = '';
+      email.isValid = true;
+    }
+
+    if (!logindata.password) {
+      password.Message = 'The Password is Required';
+      password.isValid = false;
+    } else if (logindata.password.length < password.minLength) {
+      password.Message = `The Password must contains at least ${password.minLength} chart `;
+      password.isValid = false;
+    } else {
+      password.Message = '';
+      password.isValid = true;
+    }
+
+    if (!email.isValid || !password.isValid) {
+      this.setState({ email: email, password: password });
+      return;
+    }
+
+    this.setState({ email: email, password: password, isLoading: true });
+
+    axios.post(`/user/login/`, logindata).then((req) => {
+      const datalog = req.data;
+
+      if (datalog.status !== 200) {
```

Comment:
Have you tested this? I thought that axios throws errors for non-2xx status codes. I think if the server responds 404, for example, then this code wouldn't be triggered, you'd need a catch block on your axios.post call

Reply:
in server i return  status 200 and in body of response  i return this object 
like this 
```
NotFoundMessage: (data, messag) => ({
    status: 404,
    messag,
    data,
  })
```
you can find file here [responseMessage](https://github.com/GSG-K3/event-booker/blob/master/server/helpers/responseMessage.js)

i return the status inside the object ,i handle this in client , because when i returned status 404 , 500 , ...... the axios throw error  and the app stopped , but when i use this way the app continue running .
handle !== 200 is not best practices but i use this to focus on main feature in app 

actually i can handle the status Error in catch , but i don't know What is the best practices

Reply:
Great!